src/Microsoft.OData.Core/UriParser/SemanticAst/SelectExpandClauseExtensions.cs

```diff
@@ -86,7 +86,37 @@ internal static void GetSelectExpandPaths(this SelectExpandClause selectExpandCl
         /// <returns>String list generated from selected items</returns>
         internal static List<string> GetCurrentLevelSelectList(this SelectExpandClause selectExpandClause)
         {
-            return selectExpandClause.SelectedItems.Select(GetSelectString).Where(i => i != null).ToList();
+            List<string> levelSelectList = new List<string>();
+          
+            foreach (var selectItem in selectExpandClause.SelectedItems)
+            {
+                WildcardSelectItem wildcardSelect = selectItem as WildcardSelectItem;
+                if (wildcardSelect != null)
+                {
+                    levelSelectList.Add("*");
+                    continue;
+                }
+
+                NamespaceQualifiedWildcardSelectItem namespaceQualifiedWildcard = selectItem as NamespaceQualifiedWildcardSelectItem;
+                if (namespaceQualifiedWildcard != null)
+                {
+                    levelSelectList.Add(namespaceQualifiedWildcard.Namespace + ".*" );
+                    continue;
+                }
+
+                PathSelectItem pathSelectItem = selectItem as PathSelectItem;
+                if (pathSelectItem != null)
+                {
+                    IList<string> pathSelectItems = GetSelectStringFromPathSelectItem(pathSelectItem);
+
+                    for(int i=0; i< pathSelectItems.Count; i++)

```

Comment:
>i=0 [](start = 28, length = 3)

add white space before and after "="

Reply:
It seems you can call "AddRange"?

---
In reply to: [364012742](https://github.com/OData/odata.net/pull/1607#discussion_r364012742) [](ancestors = 364012742)

Reply:
This is IList, for Addrange, it has to be concrete, so we have to make it ToList() which impact perf.
Plus Had a talk with Kanish on performance of Add vs AddRange, with AddRange being less performant. But for EF AddRange , I read they had updated their algo recently to make it better
#Resolved